neo/SmartContract/Native/NativeContract.cs

```diff
@@ -95,32 +109,32 @@ protected virtual long GetPriceForMethod(string method)
             return 0;
         }
 
-        protected virtual StackItem Main(ApplicationEngine engine, string operation, VMArray args)
-        {
-            switch (operation)
-            {
-                case "onPersist":
-                    return OnPersist(engine);
-                case "supportedStandards":
-                    return SupportedStandards.Select(p => (StackItem)p).ToList();
-            }
-            throw new NotSupportedException();
-        }
-
         internal virtual bool Initialize(ApplicationEngine engine)
         {
             if (engine.Trigger != TriggerType.Application)
                 throw new InvalidOperationException();
             return true;
         }
 
+        [ContractMethod]
```

Comment:
I think that we can set the name as mandatory

Reply:
Why?

Reply:
Because we can prevent future issues like:

Question: `I call OnXXXX and it doesn't work`
Answer: `Is onXXX`

😆

Reply:
It automatically set the method names in the ABI, so it won't be a problem.